test/functional/lua/api_spec.lua

```diff
@@ -158,13 +158,6 @@ describe('luaeval(vim.api.…)', function()
   end)
 
   it('errors out correctly when working with API', function()
-    -- Conversion errors
-    eq('Vim(call):E5108: Error executing lua [string "luaeval()"]:1: Cannot convert given lua type',
```

Comment:
shouldn't this still fail at the `luaeval` layer? if `nvim__id` now returns a lua value you put in, don't we want to test that instead?

Reply:
The second one (the one passing a dict-style table to `nvim__id`) should, and I've restored that test. And I moved the ones that now work (passing a function and passing a list containing a function) into the other test https://github.com/neovim/neovim/pull/16752/commits/67663ca0cdac302638c4cc2fecf6c233b27b327d.